src/parser/content-parser.php

```diff
@@ -114,13 +115,22 @@ protected function should_block_be_included( WP_Block $block, array $filter_opti
 	 * @return array|WP_Error
 	 */
 	public function parse( $post_content, $post_id = null, $filter_options = [] ) {
+		global $post;
+
 		Analytics::record_usage();
 
 		if ( isset( $filter_options['exclude'] ) && isset( $filter_options['include'] ) ) {
 			return new WP_Error( 'vip-block-data-api-invalid-params', 'Cannot provide blocks to exclude and include at the same time', [ 'status' => 400 ] );
 		}
 
-		$this->post_id  = $post_id;
+		// Temporarily set global $post.
+		$previous_global_post = $post;
```

Comment:
Making sure I understand this, why is this needed? Would be worth documenting the reasoning behind this.

Reply:
Improved the comment in https://github.com/Automattic/vip-block-data-api/pull/75/commits/0c1b48a52c76d18ed89d998917712dcc36f15ebb, thanks!